performance/admin.py

```diff
@@ -0,0 +1,404 @@
+from django.contrib import admin
+from io import StringIO
+from performance import models
+from django_q.tasks import async_task
+
+from frictionless import validate
+
+import os
+import csv
+
+VALID_REPORT_TYPES = ['Provincial Institutions Oversight Performance  Report',
+                      'National Institutions Oversight Performance  Report']
+
+
+def generate_import_report(report_type_validated, frictionless_report, not_matching_departments):
+    report = ""
+    if not report_type_validated:
+        report += "Report preamble must be for one of " + os.linesep
+        for report_type in VALID_REPORT_TYPES:
+            report += f"* {report_type} {os.linesep}"
+
+    if frictionless_report:
+        if not frictionless_report.valid:
+            for error in frictionless_report.tasks[0].errors:
+                report += f"* {error.message} {os.linesep}"
+
+    if len(not_matching_departments) > 0:
+        report += "Department names that could not be matched on import : " + os.linesep
+        for department in not_matching_departments:
+            report += f"* {department} {os.linesep}"
+
+    return report
+
+
+def save_imported_indicators(parsed_data, obj_id):
+    report_departments = set([x['Institution'] for x in parsed_data])
+    num_imported = 0
+    total_record_count = len(parsed_data)
+    not_matching_departments = []
+
+    for department in report_departments:
+        department_obj = models.Department.objects.filter(name=department).first()
```

Comment:
this should match zero or one department. If is are more than one match, you're picking an arbitrary one and quite likely the wrong one.

You should match the department based on

- financial year

![2023-01-10_12-46](https://user-images.githubusercontent.com/235801/211531276-79bf4e84-9a67-4137-82ae-2481f4f7bd3a.png)

- sphere (National, Provincial)

![2023-01-10_12-47](https://user-images.githubusercontent.com/235801/211531406-5961f4de-9398-4260-bed5-73855f44e445.png)

- government 

![2023-01-10_12-48](https://user-images.githubusercontent.com/235801/211531570-39fb58d6-f2c6-4aad-9e22-cc1a6a99239b.png)

- and department name as you have it

Reply:
@emre2038

Reply:
government name for national files should be `South Africa`

Reply:
done